tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TspTool/TspTool.cs

```diff
@@ -0,0 +1,340 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.ComponentModel;
+using Azure.Sdk.Tools.Cli.Services;
+using Azure.Sdk.Tools.Cli.Contract;
+using Azure.Sdk.Tools.Cli.Models.Responses;
+using ModelContextProtocol.Server;
+using Azure.Sdk.Tools.Cli.Helpers;
+
+namespace Azure.Sdk.Tools.Cli.Tools.TspTool
+{
+    /// <summary>
+    /// This tool provides functionality for initializing TypeSpec projects and converting existing Azure service swagger definitions to TypeSpec projects.
+    /// Use this tool to onboard to TypeSpec for new services or convert existing services.
+    /// </summary>
+    [McpServerToolType, Description("Tools for initializing TypeSpec projects and converting existing Azure service swagger definitions to TypeSpec projects.")]
+    public class TypeSpecTool(INpxHelper npxHelper, ILogger<TypeSpecTool> logger, IOutputService output) : MCPTool
+    {
+
+        // commands
+        private const string initCommandName = "init";
+        private const string convertSwaggerCommandName = "convert-swagger";
+
+        private readonly Argument<string> templateArg = new("--template", "The template to use for the TypeSpec project. Valid values are: azure-core (for data-plane services), azure-arm (for resource-manager services)");
```

Comment:
I think these should be defined as [Option](https://learn.microsoft.com/en-us/dotnet/api/system.commandline.option?view=net-9.0-pp) types not Argument types. Arguments are more for positional values vs. options which are flags followed by values in any order. 

Example: https://github.com/Azure/azure-sdk-tools/blob/main/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/AzurePipelines/PipelineAnalysisTool.cs#L60-L65